upt_macports/upt_macports.py

```diff
@@ -176,6 +180,47 @@ def create_package(self, upt_pkg, output=None):
             pkg_cls = pkg_classes[upt_pkg.frontend]
         except KeyError:
             raise upt.UnhandledFrontendError(self.name, upt_pkg.frontend)
-
         packager = pkg_cls()
         packager.create_package(upt_pkg, output)
+
+    def pkg_exist(self):
+        prefix = {
+            'pypi': 'py-',
+            'rubygems': 'rb-',
+            'cpan': 'p5-',
+        }
+        port_name = prefix.get(self.upt_pkg.frontend, '') + self.upt_pkg.name
+        self.logger.info(f'Checking MacPorts tree for port {port_name}')
+        cmd = f'port info --version {port_name}'
+        port = subprocess.getoutput(cmd)
+        self.logger.info(f'Current Upstream version is {self.upt_pkg.version}')
+        if port[:5] == 'Error':
+            self.logger.info(f'{port_name} not found in MacPorts tree')
+            self.curr_ver = ''
+            return False
+        else:
+            self.curr_ver = port.split()[1]
+            self.logger.info(
+                f'Current MacPorts Version for {port_name} is {self.curr_ver}')
+            return True
+
+    def check_needed(self, upt_pkg, constraint):
+        self.upt_pkg = upt_pkg
+        if self.pkg_exist():
+            curr_ver = self.curr_ver
+            upstream_ver = self.upt_pkg.version
+            # TODO: Add Decision Logic
+            if constraint:
+                self.logger.info(f'Constraint is {constraint}')
+            else:
+                self.logger.info('No constraint')
+            if curr_ver < upstream_ver:
+                self.logger.info(
+                    'Creating Portfile as new version is available')
```

Comment:
None of these error messages are really useful if you do not include the name of the package.